tokio/src/time/delay.rs

```diff
@@ -15,14 +15,19 @@ use std::task::{self, Poll};
 ///
 /// Canceling a delay is done by dropping the returned future. No additional
 /// cleanup work is required.
-pub fn delay_until(deadline: Instant) -> Delay {
+pub fn sleep_until(deadline: Instant) -> Delay {
     let registration = Registration::new(deadline, Duration::from_millis(0));
     Delay { registration }
 }
 
+/// Waits until `deadline` is reached. Alias for [`sleep_until`](sleep_until).
+pub fn delay_until(deadline: Instant) -> Delay {
```

Comment:
The main branch of Tokio is targetting 0.3, _not_ 0.2. This means that this aliasing isn't strictly necessary _unless_ you want to backport this to 0.2 (personally, I think that's a good idea, but I'll let others weigh in). If this _is_ backported this to 0.2, then I'd add a `#[deprecated]` attribute to `delay_until`  with the message "delay_until is deprecated and will be removed in Tokio 0.3; use sleep_until instead."

Reply:
I like the idea of removing them in v0.3 and having it backported to v0.2 with a `#[deprecated]` on the `delay` versions.

Reply:
Yeah, It just wasn't clear to me from the linked issue what should happen to `delay_until` and `delay_for`. My reasoning is that is doesn't really hurt to keep  both versions in 0.3 and beyond but I'm happy either way.

Reply:
Ok, so we'll go with the deprecation notice. This means I'll need to change every internal use of these two functions right? @davidbarsky @Darksonn

Reply:
If we switch to `sleep_*` fns, I don't want to maintain an alias.

Regarding backporting to 0.2, I don't have a strong opinion either way. If it were me, I probably wouldn't bother. If others think it is valuable, I will punt to them.